### PR TITLE
Clean up relationship between deployment_tar and deployment

### PR DESCRIPTION
**Pull request by bfineran**
## Feature Description

Now, any directory that can be downloaded as tar.gz and then unzipped is by default downloaded in such a manner. 
This does not prevent us from downloading single, "loose" files from the directory in question.

### Downloading the `deployment` directory
Note, that the deployment is downloaded as tar and then unzipped:

```python
stub = "zoo:mobilenet_v1-1.0-imagenet-pruned.4block_quantized"

model = Model(stub)
deployment_path = model.deployment.path
print(os.listdir(deployment_path))
print(os.listdir(model._path))
```

```bash
Downloading (…)ed/deployment.tar.gz: 100%|██████████| 4.48M/4.48M [00:00<00:00, 12.7MB/s]
['model.onnx']
['deployment', 'deployment.tar.gz']
```

### Downloading the whole model
Note, that the appropriate folders get downloaded as tars and then unzipped

```python
model = Model(stub)
print(os.listdir(model.path))
```

```bash
Downloading (…)d/training/model.pth: 100%|██████████| 32.6M/32.6M [00:02<00:00, 12.5MB/s]
Downloading (…)training-metric.yaml: 100%|██████████| 168/168 [00:00<00:00, 89.7kB/s]
Downloading (…)ed/deployment.tar.gz: 100%|██████████| 4.48M/4.48M [00:00<00:00, 12.0MB/s]
Downloading (…)ple-originals.tar.gz: 100%|██████████| 4.07M/4.07M [00:00<00:00, 9.58MB/s]
Downloading (…)sample-inputs.tar.gz: 100%|██████████| 3.31M/3.31M [00:00<00:00, 11.1MB/s]
Downloading (…)ample-outputs.tar.gz: 100%|██████████| 152k/152k [00:00<00:00, 4.29MB/s]
Downloading (…)sample-labels.tar.gz: 100%|██████████| 170k/170k [00:00<00:00, 3.43MB/s]
Downloading (…)k_quantized/model.md: 100%|██████████| 1.98k/1.98k [00:00<00:00, 1.19MB/s]
Downloading (…)ed/model.onnx.tar.gz: 100%|██████████| 4.48M/4.48M [00:00<00:00, 10.6MB/s]
['sample-inputs', 'sample-labels.tar.gz', 'deployment', 'sample-originals', 'sample-originals.tar.gz', 'deployment.tar.gz', 'model.md', 'sample-outputs.tar.gz', 'sample-inputs.tar.gz', 'model.onnx', 'sample-outputs', 'training', 'model.onnx.tar.gz', 'sample-labels']
```

### Downloading a single "loose" `deployment` file

```python
model = Model(stub)
deployment_onnx_file = model.deployment.get_file("model.onnx").path
print(os.listdir(os.path.dirname(deployment_onnx_file)))
print(os.listdir(model._path))
```

```bash
Downloading (…)eployment/model.onnx: 100%|██████████| 7.00M/7.00M [00:00<00:00, 11.4MB/s]
['model.onnx']
['deployment']
```

### CODE REVIEW

**Comment by dbogunowicz** [comment]
Reminder: once this PR lands, we need to update the relevant pathways to the `deployment` directory in sparseml and deepsparse.

File: src/sparsezoo/objects/directory.py [hunk 1]
```diff
@@ -167,8 +167,13 @@ def download(
                     "Failed to recognize a valid download path. "
                     "Please make sure that `destination_path` argument is not None."
                 )
+        if getattr(self, "tar_directory", None) is not None:
+            # if tar_directory is not None, then we are downloading
+            # the directory as a tar archive file
+            self = self.tar_directory
```
Comment (bfineran) [inline src/sparsezoo/objects/directory.py:173]: don't like how we are weakly overriding the self variable here

Comment (bfineran) [reply src/sparsezoo/objects/directory.py:173]:
```python
target_directory = self if getattr(self, "tar_directory", None) is None else self.tar_directory
```

File: src/sparsezoo/model/model.py [hunk 2]
```diff
@@ -636,8 +612,13 @@ def _directory_from_files(
...
         display_name: Optional[str] = None,
-        regex: Optional[bool] = False,
-        allow_multiple_outputs: Optional[bool] = False,
+        regex: bool = False,
+        allow_multiple_outputs: bool = False,
+        # allow picking one from multiple directories
+        # is used when we initialize the model from
+        # local directory (not stub) and both [some_directory]
+        # and [some_directory].tar.gz are present
+        allow_picking_one_from_multiple: bool = False,
```
Comment (bfineran) [inline src/sparsezoo/model/model.py:621]: needs better name

Comment (bfineran) [reply src/sparsezoo/model/model.py:621]: like download_from_tar_if_available

**Review by dbogunowicz — APPROVED** [review]
best pr ever

**Pull request merged**